integration/spark/src/test/scala/org/apache/carbondata/spark/testsuite/datacompaction/CarbonIndexFileMergeTestCase.scala

```diff
@@ -530,21 +615,28 @@ class CarbonIndexFileMergeTestCase
       FileFactory.getCarbonFile(table.getAbsoluteTableIdentifier.getTablePath)
         .listFiles(true, new CarbonFileFilter {
           override def accept(file: CarbonFile): Boolean = {
-            file.getName.endsWith(extension)
+            file.getName.endsWith(CarbonTablePath.INDEX_FILE_EXT) ||
+            file.getName.endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT)
           }
         })
     } else {
       FileFactory.getCarbonFile(path).listFiles(true, new CarbonFileFilter {
         override def accept(file: CarbonFile): Boolean = {
-          file.getName.endsWith(extension)
+          file.getName.endsWith(CarbonTablePath.INDEX_FILE_EXT) ||
+          file.getName.endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT)
         }
       })
     }
-    if (carbonFiles != null) {
-      carbonFiles.size()
-    } else {
-      0
-    }
+    var validIndexFiles = SegmentFileStore.getValidCarbonIndexFiles(carbonFiles.asScala.toArray)
+    validIndexFiles = validIndexFiles.toStream
+      .filter(file => file.getName.endsWith(extension)).toArray
+    validIndexFiles.length
+  }
+
+  def getSegmentFileCount(tableName: String): Int = {
```

Comment:
please check if similar code is written in other test cases, if so, you can move to some test util.